proposals/0014-wave-size-range.md

```diff
@@ -0,0 +1,283 @@
+  The third parameter may also be zero.
+* The minimum wave size is less than the maximum wave size.
+* The preferred wave size lies between the minimum and maximum.
+
+### Runtime Additions
+
+#### Runtime information
+
+The PSV0 runtime data structure already contains both
+ `MinimumExpectedWaveLaneCount` and `MaximumExpectedWaveLaneCount` members that
+ can be used to transmit the minimum and maximum values to the runtime.
+A new `PreferredWaveLaneCount` value will need to be added to the PSV3 revision
+ of that struct to accommodate the third preferred value.
+
+#### Device Capability
+
+As a required feature, devices supporting the shader model it ships with are
+ required to respect the wave size restrictions indicated by the wave size range
+ metadata.
+
+## Testing
+
+### Correct Behavior Testing
+
+Verify this compiler output:
+
+* The two parameter overload of `WaveSize` correctly transmits those values to a
+  metadata tuple with a zero third value that is pointed to by the correct tag
+  in the entry point attribute list.
+* The three parameter overload of `WaveSize` transmits those values as well as
+  the third in the same tuple.
+* That the PSV0 `MinimumExpectedWaveLaneCount` and `MaximumExpectedWaveLaneCount`
+  values reflect those provided for the wave size range.
+* That the PSV3 `PreferredWaveLaneCount` value reflects the third preferred
+  value when provided and zero when omitted.
+
+#### Diagnostics Testing
+
+* Use the following invalid parameters each parameter location to `WaveSize`
+  and ensure that an appropriate error is produced:
+  * Negative power of two integer
+  * Floating point value
+  * non-literal
+  * Integer less than 4
+  * Integer greater than 128
+  * A non-power-of-two integer between 4 and 128
+* Add the following invalid `WaveSize` attributes to an compute shader entry
+  point and ensure that an appropriate error is produced:
+  * no parameter list
+  * an empty parameter list "()"
+  * four parameters
+* Try the following invalid `WaveSize` parameter value combinations and ensure
+  that an appropriate error is produced:
+  * Set the minimum wave size equal to the maximum
+  * Set the minimum wave size greater than the maximum
+  * Set the preferred wave size to a value outside of the specified range
+* Combine multiples of the 1, 2 and 3 parameter `WaveSize` attribute overloads
+  on the same entry point and ensure that an attribute conflict error is
+  produced.
+
+### Validation Testing
+
+Test that the following produce validation errors:
+
+* The wave size range tag pointing to anything but a tuple of 3
+* A tuple value is not an integer
+* A range tuple value is -4, 0, 1, 2, 3, 127, 129, or 256
+* A preferred tuple value is -4, 1, 2, 3, 127, 129, or 256
+* The minimum wave size value is equal to the maximum
+* The minimum wave size value is greater than the maximum
+* The preferred wave size is outside the specified range, but otherwise valid
+
+### Execution Testing
```

Comment:
Some of these validations sound a lot like HLKs.  Would we normally call out new HLKs as part of this proposal?